cleanrl_utils/huggingface.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import os
+import sys
+from pathlib import Path
+from pprint import pformat
+from typing import List
+
+import numpy as np
+
+HUGGINGFACE_VIDEO_PREVIEW_FILE_NAME = "replay.mp4"
+HUGGINGFACE_README_FILE_NAME = "README.md"
+
+
+def push_to_hub(
+    args: argparse.Namespace,
+    episodic_returns: List,
+    repo_id: str,
+    algo_name: str,
+    folder_path: str,
+    video_folder_path: str = "",
+    revision: str = "main",
+    create_pr: bool = False,
+    private: bool = False,
+):
+    # Step 1: lazy import and create / read a huggingface repo
+    from huggingface_hub import CommitOperationAdd, CommitOperationDelete, HfApi
+    from huggingface_hub.repocard import metadata_eval_result, metadata_save
+
+    api = HfApi()
+    repo_url = api.create_repo(
+        repo_id=repo_id,
+        exist_ok=True,
+        private=private,
+    )
+    # parse the default entity
+    entity, repo = repo_url.split("/")[-2:]
+    repo_id = f"{entity}/{repo}"
+
+    # Step 2: clean up data
+    # delete previous tfevents and mp4 files
+    operations = [
+        CommitOperationDelete(path_in_repo=file)
+        for file in api.list_repo_files(repo_id=repo_id)
+        if file.endswith(".tfevents") or file.endswith(".mp4")
+    ]
+
+    # Step 3: Generate the model card
+    algorith_variant_filename = sys.argv[0].split("/")[-1]
+    model_card = f"""
+# (CleanRL) **{algo_name}** Agent Playing **{args.env_id}**
+
+This is a trained model of a {algo_name} agent playing {args.env_id}.
+The model was trained by using [CleanRL](https://github.com/vwxyzjn/cleanrl) and the most up-to-date training code can be
+found [here](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/{args.exp_name}.py).
+
+## Command to reproduce the training
+
+```bash
+curl -OL https://huggingface.co/{repo_id}/raw/main/dqn.py
+curl -OL https://huggingface.co/{repo_id}/raw/main/pyproject.toml
+curl -OL https://huggingface.co/{repo_id}/raw/main/poetry.lock
+poetry install --all-extras
+python {algorith_variant_filename} {" ".join(sys.argv[1:])}
+```
+
+# Hyperparameters
+```python
+{pformat(vars(args))}
+```
+    """
+    readme_path = f"{folder_path}/README.md"
```

Comment:
```suggestion
    readme_path = Path(folder_path) / HUGGINGFACE_README_FILE_NAME
```
Here it would break for your Windows users. Also you can use the `HUGGINGFACE_README_FILE_NAME` variable you defined above :)